tests/utils/palette_review.py

```diff
@@ -0,0 +1,96 @@
+         '## Notes\n'
+         '- ___keywords___: bg, background, fg, foreground, color, matplotlib '
+         'color names\n'
+         '- Currently assumes all non-background colors are foreground colors for the test)\n')
+legend = ("## Legend\n"
+          "- ✅: all WCAG tests passed\n"
+          "- ☑️: some WCAG tests passed\n"
+          "- ❌: no WCAG tests passed\n"
+          "- Symbols link to WCAG api response with tests")
+
+with open(Path(os.getcwd()) / "tests" / "palette_report.md", "w") as f:
+    f.write(intro)
+    f.write(legend)
+for p in palettes:
+    backgrounds = {}
+    foregrounds = {}
+    sheet = cssutils.parseFile(p)
+    for rule in sheet:
+        if isinstance(rule, cssutils.css.csscomment.CSSComment):
+            continue
+        style_text = rule.style.cssText.replace("\n", "")
+        for i in style_text.split(";"):
+            pair = i.split(": ")
+            label = pair[0].replace("--", "")
+            value = pair[1].replace("#", "")
+            if "invert" in label or "var(" in value:
+                pass
+            else:
+                if value == "fff":
+                    value = "ffffff"
+                if any(x in label for x in bg_labels):
+                    backgrounds[label] = value
+                else:
+                    if any(x in label for x in fg_labels):
+                        foregrounds[label] = value
+                    elif any(x in label.replace("-", "") for x in list(color_labels)):   
+                        foregrounds[label] = value
+                    elif "color" in label:
+                        foregrounds[label] = value
+    if len(backgrounds.keys()) < 1 or len(foregrounds.keys()) < 1:
+        with open(Path(os.getcwd()) / "tests" / "palette_report.md", "a+") as f:
+            f.write(f"\n# {p.name}")
+            f.write("\n- Style does not have enough foregrounds and bakcgrounds to compare.\n")
+    else:
+        backgrounds = {k: v for k, v in sorted(backgrounds.items(), key=lambda item: item[1])}
+        foregrounds = {k: v for k, v in sorted(foregrounds.items(), key=lambda item: item[1])}
+        table_start = "\n<table>\n  <thead>\n    <tr>"
+        headers = "\n      <th>backgrounds</th>"
+        table_middle = "\n  <tbody>"
+        table_end = "\n  </tbody>\n</table>"
+        for k, v in foregrounds.items():
+            headers += f"\n      <th>{k} ({v})</th>"
+        for k, v in backgrounds.items():
+            row = f"\n    <tr>\n      <td>{k} ({v})</td>"
+            for x, y in foregrounds.items():
+                try:
+                    url = f"https://webaim.org/resources/contrastchecker/?fcolor={y}&bcolor={v}&api"
+                    resp = get(url)
+                    grades = resp.json()
+                    del grades["ratio"]
+                    if all(grade == "pass" for grade in grades.values()):
+                        row += f"\n      <td><a href='{url}'>:white_check_mark:</a></td>"
```

Comment:
can we strip the `&api` from the href here so that the link brings you to the [regular browser result page](https://webaim.org/resources/contrastchecker/?fcolor=fb4934&bcolor=504945) instead of the JSON output?

ie `<a href='https://webaim.org/resources/contrastchecker/?fcolor=fb4934&bcolor=504945'>`

Reply:
Done!

Reply:
ty!